app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/tiles/GeneratorTileService.kt

```diff
@@ -0,0 +1,53 @@
+package com.x8bit.bitwarden.ui.platform.feature.tiles
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.service.quicksettings.TileService
+import com.x8bit.bitwarden.ui.platform.manager.intent.IntentManager
+import com.x8bit.bitwarden.ui.platform.manager.intent.IntentManagerImpl
+import kotlinx.coroutines.Runnable
+
+/**
+ * A service for handling the Password Generator quick settings tile.
+ */
+class GeneratorTileService : TileService() {
+    private val tileIntentManager: IntentManager = IntentManagerImpl(this)
+
+    override fun onTileAdded() {
+        super.onTileAdded()
+    }
+
+    override fun onStartListening() {
+        super.onStartListening()
+    }
+
+    override fun onStopListening() {
+        super.onStopListening()
+    }
+
+    override fun onTileRemoved() {
+        super.onTileRemoved()
+    }
+
+    override fun onClick() {
+        super.onClick()
+
+        if (isLocked) {
+            unlockAndRun(Runnable { launchGenerator(); })
+        } else {
+            launchGenerator()
+        }
+    }
+
+    @Suppress("DEPRECATION")
+    @SuppressLint("StartActivityAndCollapseDeprecated")
+    private fun launchGenerator() {
+        val intent = tileIntentManager.createTileIntent("bitwarden://password_generator")
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            startActivityAndCollapse(tileIntentManager.createTilePendingIntent(intent))
+        } else {
+            startActivityAndCollapse(intent)
```

Comment:
Of course these are members of TileService, so we can't create a Processor and write tests. 😢